File: wasm/test/docs/test_vertex_attr.js
import { describe, it } from 'mocha';
import {DirectedGraphHierarchicalLayout, AttributeRankDir, AttributeShape, buildVectorString} from '../../index.js';
import { compareSVG } from './utils.js';

describe('DocsVertexAttributes', () => {
    it('shape', async () => {
        const layout = new DirectedGraphHierarchicalLayout();
        const graph = layout.createGraph(3);
        graph.addEdges([[0, 1], [1, 2], [2, 0]]);
        layout.attributes().setVertexShape(0, AttributeShape.DOUBLE_CIRCLE);
        layout.attributes().setVertexShape(1, AttributeShape.RECT);
        layout.attributes().setVertexShape(2, AttributeShape.ELLIPSE);
        layout.attributes().setRankDir(AttributeRankDir.LEFT_TO_RIGHT);
        layout.initVertexLabelsWithNumericalValues(0);
        layout.layoutGraph();
        const svg = layout.render();
        await compareSVG("vertex_attr", "shape", svg);
    });
    it('label', async () => {
        const layout = new DirectedGraphHierarchicalLayout();
        const graph = layout.createGraph(3);
        graph.addEdges([[0, 1], [1, 2], [2, 0]]);
        layout.setVertexLabels(buildVectorString(["Eat", "Sleep", "Play"]));
        for (let u = 0; u < 3; ++u) {
            layout.attributes().setVertexShape(u, AttributeShape.ELLIPSE);
        }
        layout.attributes().setRankDir(AttributeRankDir.LEFT_TO_RIGHT);
        layout.layoutGraph();
        const svg = layout.render();
        await compareSVG("vertex_attr", "label", svg);
    });
});

File: python/tests/docs/test_vertex_attr.py
from sp_graph_layout import (
    AttributeRankDir,
    AttributeShape,
    DirectedGraphHierarchicalLayout,
)

from .utils import compare_svg


def test_shape():
    layout = DirectedGraphHierarchicalLayout()
    graph = layout.create_graph(3)
    graph.add_edges([[0, 1], [1, 2], [2, 0]])
    layout.attributes().set_vertex_shape(0, AttributeShape.DOUBLE_CIRCLE)
    layout.attributes().set_vertex_shape(1, AttributeShape.RECT)
    layout.attributes().set_vertex_shape(2, AttributeShape.ELLIPSE)
    layout.attributes().set_rank_dir(AttributeRankDir.LEFT_TO_RIGHT)
    layout.init_vertex_labels_with_numerical_values(0)
    layout.layout_graph()
    svg = layout.render()
    compare_svg("vertex_attr", "shape", svg)


def test_label():
    layout = DirectedGraphHierarchicalLayout()
    graph = layout.create_graph(3)
    graph.add_edges([[0, 1], [1, 2], [2, 0]])
    layout.set_vertex_labels(["Eat", "Sleep", "Play"])
    for u in range(3):
        layout.attributes().set_vertex_shape(u, AttributeShape.ELLIPSE)
    layout.attributes().set_rank_dir(AttributeRankDir.LEFT_TO_RIGHT)
    layout.layout_graph()
    svg = layout.render()
    compare_svg("vertex_attr", "label", svg)
